### ISSUE TITLE
AI should prioritize attacking units, not buildings

### ISSUE THREAD

**Issue by andy5995** [ZetaGlest/zetaglest-source#54]
AI units should not attack a totem, for example, while it's being attacked by an enemy. If AI units are being attacked by units (mobile or otherwise), it should attack those first, and then attack buildings and other units.

This would be one step to making the AI smarter.

**Comment by andy5995** [comment]
Yes, and if you do, join our Discord channel https://discord.gg/WaAaXS7

**Comment by biels** [comment]
I would go one step further and target units that have the best score
calculated as follows:
EnemyAttackDamage*EnemyAttackSpeed*EnemySplashRadius /
(EnemyRemainingHealth*EffectiveDamageMultiplier + 1)


This "heuristic" can be improved a lot but I think it is a great place to
start since it is quite easy to calculate.

For example not taking into account splash radius if we don't have ally
units close to us, not targeting units that are already going to die and
slightly biasing the formula towards attacking units that are further so we
make better use of our range. Also maybe we can calculate if we are going
to miss all the time by looking at the projectile speed and enemy movement
speed and lower its value so we only attack it if we have nothing better to
do and hope it stops or does a sharp turn.

**Comment by andy5995** [comment]
Yeah, that's a good idea, but it needs to be in a separate ticket.
